packages/algorand-js/src/interpreter/interpreter.ts

```diff
@@ -1,35 +1,85 @@
+import { mkTransaction } from "algob";
 import type { execParams } from "algob/src/types";
+import { AccountInfo, assignGroupID, Transaction } from "algosdk";
 import { assert } from "chai";
 
+import { mockSuggestedParams } from "../../test/mocks/txn";
 import { TealError } from "../errors/errors";
 import { ERRORS } from "../errors/errors-list";
 import { DEFAULT_STACK_ELEM } from "../lib/constants";
 import { Stack } from "../lib/stack";
-import type { Operator, StackElem, TEALStack } from "../types";
+import type { AccountsMap, Operator, StackElem, TEALStack } from "../types";
 
 export class Interpreter {
   readonly stack: TEALStack;
   bytecblock: Uint8Array[];
   intcblock: BigInt[];
   scratch: StackElem[];
+  tx: Transaction;
+  gtxs: Transaction[];
+  accounts: AccountsMap;
 
   constructor () {
     this.stack = new Stack<StackElem>();
     this.bytecblock = [];
     this.intcblock = [];
     this.scratch = new Array(256).fill(DEFAULT_STACK_ELEM);
+    this.accounts = <AccountsMap>{};
+    this.tx = <Transaction>{}; // current transaction
+    this.gtxs = []; // all transactions
   }
 
   /**
- * Description: this function executes set of Operator[] passed after
- * parsing teal code
- * @param {execParams} txn : Transaction parameters for current txn
- * @param {Logic[]} logic : set of Logic objects
- * @param {AppArgs} args : argument array passed to the current transaction
- * @returns {boolean} : transaction accepted/rejected based on asc logic
- */
-  execute (txn: execParams, logic: Operator[], args: Uint8Array[]): boolean {
+   * Description: creates a new transaction object from given execParams
+   * @param txnParams : Transaction parameters for current txn or txn Group
+   */
+  createTxnContext (txnParams: execParams | execParams[]): void {
+    if (Array.isArray(txnParams)) {
+      if (txnParams.length > 16) {
+        throw new Error("Maximum size of an atomic transfer group is 16");
+      }
+
+      const txns = [];
+      for (const txnParam of txnParams) {
+        const mockParams = mockSuggestedParams(txnParam.payFlags);
+        const txn = mkTransaction(txnParam, mockParams);
+        txns.push(txn);
+      }
+      assignGroupID(txns);
+      this.gtxs = txns;
+      this.tx = txns[0]; // by default current txn is the first txn
+    } else {
+      const mockParams = mockSuggestedParams(txnParams.payFlags);
+      this.tx = mkTransaction(txnParams, mockParams);
+      this.gtxs = [this.tx];
+    }
```

Comment:
is the `else` part for tests? 
Hacking a normal code for tests execution is usually a pattern. Why not passing mock params through an argument to this function?

Reply:
The `else` part is not for tests. It represents a single transaction (not a transaction group).
`mock Params` are not passed explicitly because some params are dynamic (eg `txn fee`). 
And for an atomic group, `mock Params` might be different for different transactions (eg - different `fee` / `firstValid` /..etc)

Reply:
If the `txnParams` is an Array then it is representing a transaction group (Multiple transactions as an atomic group). So each object in that array represents a single transaction.

Otherwise (the `else` part), we create and assign a single transaction.

Reply:
I meant to include this explanation in the code next time you will do a PR, ;)